### ISSUE TITLE
[Bug] Got an error when updating unsupported fields of pod

### ISSUE THREAD

**Issue by whitewindmills** [karmada-io/karmada#3525]
**What happened**:
After propagating a pod to some member clusters, I'll get an error if I try to update some unsupported fields of pod.
```
error: Pod "demo" is invalid: spec: Forbidden: pod updates may not change fields other than `containers[*].image` or `spec.activeDeadlineSeconds`
```

**What you expected to happen**:
Unsupported fields of pod should be retained.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:
Of course, only a few fields support being updated, in place.
https://kubernetes.io/docs/concepts/workloads/pods/#pod-update-and-replacement

**Environment**:
- Karmada version: all version
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by chaunceyjiang** [comment]
> error: Pod "demo" is invalid: spec: Forbidden: pod updates may not change fields other than `containers[*].image` or `spec.activeDeadlineSeconds`

Can you give me a sample pod.yaml file  and pp.yaml  file .  How to reproduce it ?

**Comment by whitewindmills** [comment]
@chaunceyjiang 
No special restrictions except I have a controller for _Pod_. If the pod on the member cluster was modified before creating, later karmada will try to overwrite the modification, but these fields are not allowed to be modified in _K8s_, which is the reason for this error.

**Comment by chaunceyjiang** [comment]
`Karmada` reuses `kube-apiserver`, so these limitations should also exist in `Karmada`.

**Comment by chaunceyjiang** [comment]
I tried to modify some fields in the pod within Karmada and received the following information. 

<img width="1367" alt="image" src="https://github.com/karmada-io/karmada/assets/17962021/65c871f0-e2fc-447a-a194-73cfdcb9bbad">

**Comment by whitewindmills** [comment]
You cannot update those fields. 
> No special restrictions except I have a controller for Pod. If the pod on the member cluster was modified before creating, later karmada will try to overwrite the modification, but these fields are not allowed to be modified in K8s, which is the reason for this error.

I think you do not get me. I did not modify the pod on the Karmada control plane, but modified the pod before creating the pod on the member cluster, so Karmada will always try to overwrite this modification. Finally, you'll get this error.

**Comment by chaunceyjiang** [comment]
> modified the pod before creating the pod on the member cluster, so Karmada will always try to overwrite this modification. 

Got it . 😄

**Comment by RainbowMango** [comment]
I don't quite understand what's going on here.
> I did not modify the pod on the Karmada control plane, but modified the pod before creating the pod on the member cluster, so Karmada will always try to overwrite this modification.

How can you `modify` a Pod before `creating` it?

**Comment by RainbowMango** [comment]
I'm not sure if this is a reasonable use case. After taking a glance at #3526 and my concern is that the specification on the Karmada control plane might be different with member clusters, I doubt it's the expected behavior from a user's respective.

**Comment by whitewindmills** [comment]
> How can you `modify` a Pod before `creating` it?

As mentioned in the [issue](https://github.com/karmada-io/karmada/issues/3572), member clusters may have some standard injection behavior, which will lead to differences between Karmada control plane and member clusters. But most importantly, these fields are not allowed to be modified in K8s, so Karmada should retain these fields.

**Comment by whitewindmills** [comment]
@RainbowMango 
Regardless of whether the injection behavior is the standard of K8s, but for Pod, it is the standard of K8s that most fields are not allowed to be modified. Karmada should abide by this rule and retain them by default. So I think the [PR](https://github.com/karmada-io/karmada/pull/3526) does make sense.

**Comment by RainbowMango** [comment]
Yeah, #3572 seems a reasonable use case. Let's continue the discussion there. I want to clarify the use case thoroughly before making decisions.
